server/netty-server/sync/src/main/scala/sttp/tapir/server/netty/sync/internal/ox/OxDispatcher.scala

```diff
@@ -20,12 +23,13 @@ import ox.channels.Actor
   */
 private[sync] class OxDispatcher()(using ox: Ox):
   private class Runner:
-    def runAsync(thunk: Ox ?=> Unit, onError: Throwable => Unit): Unit =
-      fork {
+    def runAsync(thunk: Ox ?=> Unit, onError: Throwable => Unit, cf: CompletableFuture[CancellableFork[Unit]]): Unit =
+      cf.complete(forkCancellable {
         try supervised(thunk)
-        catch case e => onError(e)
-      }.discard
+        catch case NonFatal(e) => onError(e)
+      }).discard
 
   private val actor = Actor.create(new Runner)
 
-  def runAsync(thunk: Ox ?=> Unit)(onError: Throwable => Unit): Unit = actor.tell(_.runAsync(thunk, onError))
+  def runAsync(thunk: Ox ?=> Unit)(forkFuture: CompletableFuture[CancellableFork[Unit]])(onError: Throwable => Unit): Unit =
```

Comment:
maybe this can simply return `Future[CancellableFork[Unit]]`? + a comment that we can't use `actor.ask` as it might be blocking

Reply:
We need to pass a "holder" (CompletableFuture in this case) from the outside, because it will be referenced in onComplete/onError of the Processor.

Reply:
well I guess you could also assign the result to a var in the processor, but that's not such a huge difference

Reply:
yeah, I guess a scala or java future doesn't matter much here really?